### DISCUSSION TITLE
Please document how to clear the cache when using WebView2 within Unity

### DISCUSSION THREAD

**Discussion by RubenGarcia** [MicrosoftEdge/WebView2Feedback#4058] [Q&A]
I tried
```
using Microsoft.Web.WebView2.Core;
CoreWebView2Profile.ClearBrowsingDataAsync();
```
which gives
```
 The type or namespace name 'Web' does not exist in the namespace 'Microsoft'`
```

**Comment by michaelfarnsworth** [comment]
@RubenGarcia our WebView Unity plugin exposes a subset of what's available with the underlying CoreWebView2 control. You can find the current API surface area here: https://learn.microsoft.com/en-us/windows/mixed-reality/develop/advanced-concepts/webview2-unity-plugin. We don't currently expose a method for clearing the cache. We can look into that.

**Comment by RubenGarcia** [comment]
Yes, clearing the cache would be very useful. 
Please expose it.
Thank you.

**Comment by michaelfarnsworth** [comment]
@RubenGarcia are you interested in clearing everything? Browsing history, cookies, temp files, etc.?

**Comment by RubenGarcia** [comment]
My usecase is being able to change the html in a webpage, and seeing the changes in WebView2. Right now, if I change the webpage, Webview2 shows the old, cached page.

**Comment by RubenGarcia** [comment]
Another thing which I would like exposed is a way to get and set the ZoomFactor.
This is especially important since wheel functionality to traverse long pages is currently not working
(https://github.com/MicrosoftEdge/WebView2Feedback/issues/3705)

**Comment by michaelfarnsworth** [comment]
> My usecase is being able to change the html in a webpage, and seeing the changes in WebView2. Right now, if I change the webpage, Webview2 shows the old, cached page.

Could you give a few more details on your scenario? I did some quick testing, but didn't hit a caching issue. Is the page hosted locally? How is the HTML modified? I assume you're refreshing and expecting the updated content?

**Comment by RubenGarcia** [comment]
I browsed
https://[rubengarcia.userweb.mwn.de/videos/](https://rubengarcia.userweb.mwn.de/videos/)
with hololens.
Then closed the program
Then used sftp to upload a new page with a different background colour to https://rubengarcia.userweb.mwn.de/videos/index.html
Then I opened the hololens again, and the page displayed the old background colour.

**Comment by michaelfarnsworth** [comment]
@RubenGarcia how are you closing the app?

**Comment by RubenGarcia** [comment]
In hololens, tap on wrist, click on the button at the bottom of the menu, then long press on app name in menu, then close.
